src/model/mod.rs

```diff
@@ -235,12 +236,94 @@ fn valid_field(player: &types::Player, input: usize, ori: &str) -> bool {
             }
         },
     }
-
     valid
 }
 
+/// Removes the current rand value from the remaining possibilities.
+/// Since every call of remove() shifts every remaining element to the left,
+/// we need to search for the index of the element to be deleted.
+fn remove_idx(rand: usize, vec: &mut Vec<usize>) {
+    for i in 0..vec.len() - 1 {
+        if vec[i] == rand {
+            vec.remove(i);
+        }
+    }
+}
+
+/// Random ship placement for the AI.
+fn place_ai(player: &mut types::Player, ship: &types::ShipType,
+    mut vec: &mut Vec<usize>) -> Result<(), types::ErrorType> {
+
+    /* Vector to collect the indices for the possible ship position. */
+    let mut indices = Vec::new();
+
+    let mut rng = thread_rng();
+    let mut rand = *rng.choose(&vec).unwrap();
+
+    /* The complete Moore neighborhood needs to be free
+       to place the first part of the ship. */
+    loop {
+        /* If every possibility was tested, we have a situation
+           in which the remaining boats can't be placed, so we
+           need to restart the whole placement process. */
+        if vec.len() == 1 {
+            return Err(types::ErrorType::DeadEnd);
+        } else if valid_field(&player, rand, &"".to_string()) {
```

Comment:
`&"".to_string()` first making `""` a String, then taking a reference of it? Why not just `""`?